### PR TITLE
lepton-eda: init at 1.9.13-20201211

### PR DESCRIPTION
**Pull request by tesq0**
###### Motivation for this change

Needed some nice software for designing electronics

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/electronics/lepton-eda/default.nix [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  shortVer = builtins.head (lib.splitString "-" version);
+in
+stdenv.mkDerivation rec {
+
+  pname = "lepton-eda";
+
+  inherit version;
+
+  src = fetchurl {
+    url = "https://github.com/lepton-eda/lepton-eda/releases/download/${version}/lepton-eda-${shortVer}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/lepton-eda/default.nix:31]:
```suggestion
stdenv.mkDerivation rec {
  pname = "lepton-eda";
  version = "1.9.13-20201211";

  src = fetchurl {
    url = "https://github.com/lepton-eda/lepton-eda/releases/download/${version}/lepton-eda-${builtins.head (lib.splitString "-" version)}.tar.gz";
```

That should work. Didn't test.

File: pkgs/applications/science/electronics/lepton-eda/default.nix [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    sha256 = "17z1r9mdc8b4q6k8x7xv8ixkbkzhrlnw4l53wn64srd72labf5zl";
+  };
+
+  nativeBuildInputs = [ pkgconfig makeWrapper texinfo autoreconfHook ];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/lepton-eda/default.nix:35]:
```suggestion
  nativeBuildInputs = [ pkg-config makeWrapper texinfo autoreconfHook ];
```

File: pkgs/applications/science/electronics/lepton-eda/default.nix [hunk 3]
```diff
@@ -0,0 +1,70 @@
+{ stdenv,
+ lib,
+ pkgconfig,
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/lepton-eda/default.nix:3]:
```suggestion
 pkg-config,
```

File: pkgs/applications/science/electronics/lepton-eda/default.nix [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  postInstall = ''
+
+    libs="${lib.makeLibraryPath propagatedBuildInputs}"
+  	for program in $out/bin/*
+    do
+        wrapProgram "$program" \
+        --prefix LD_LIBRARY_PATH : "$libs" \
+        --prefix LTDL_LIBRARY_PATH : "$out/lib"
+    done
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/lepton-eda/default.nix:56]:
```suggestion
  postInstall = ''
    libs="${lib.makeLibraryPath propagatedBuildInputs}"
    for program in $out/bin/*; do
      wrapProgram "$program" \
        --prefix LD_LIBRARY_PATH : "$libs" \
        --prefix LTDL_LIBRARY_PATH : "$out/lib"
    done
  '';
```

File: pkgs/applications/science/electronics/lepton-eda/default.nix [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    longDescription = ''
+      Lepton EDA is a suite of free software tools for designing electronics.
+      It provides schematic capture, netlisting into over 30 netlist formats, and many other features.
+    '';
+    license = licenses.gpl2;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ tesq0 ];
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/lepton-eda/default.nix:70]:
```suggestion
  };
}
```

File: pkgs/applications/science/electronics/lepton-eda/default.nix [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/lepton-eda";
+    description = "Lepton Electronic Design Automation";
+    longDescription = ''
+      Lepton EDA is a suite of free software tools for designing electronics.
+      It provides schematic capture, netlisting into over 30 netlist formats, and many other features.
+    '';
+    license = licenses.gpl2;
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/lepton-eda/default.nix:65]: gpl2Plus or gpl2Only?

File: pkgs/development/libraries/gtkextra/default.nix [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+let
+  minorVersion = "3.3";
+  microVersion = "4";
+  version = "${minorVersion}.${microVersion}";
+in
+stdenv.mkDerivation rec {
+
+  pname = "gtkextra";
+
+  inherit version;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gtkextra/default.nix:26]:
```suggestion
let
  minorVersion = "3.3";
  microVersion = "4";
  version = "${minorVersion}.${microVersion}";
in
stdenv.mkDerivation rec {
  pname = "gtkextra";
  inherit version;
```

File: pkgs/development/libraries/gtkextra/default.nix [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+  src = fetchurl {
+    url = "mirror://sourceforge/project/gtkextra/${minorVersion}/${pname}-${version}.tar.gz";
+    sha256 = "1mpihbyzhv3ymfim93l9xnxmzhwyqdba5xb4rdn5vggdg25766v5";
+  };
+
+  preConfigure = ''
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gtkextra/default.nix:33]:
```suggestion
  postPatch = ''
```

Does that still work in preConfigure or do we miss a phase?

File: pkgs/development/libraries/gtkextra/default.nix [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+    meta = with lib; {
+    homepage = "http://gtkextra.sourceforge.net/";
+    description = "GtkExtra is a useful set of widgets for creating GUI's for GTK+.";
+    license = licenses.lgpl2;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ tesq0 ];
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gtkextra/default.nix:49]:
```suggestion
  meta = with lib; {
    homepage = "http://gtkextra.sourceforge.net/";
    description = "GtkExtra is a useful set of widgets for creating GUI's for GTK+.";
    license = licenses.lgpl2;
    platforms = platforms.linux;
    maintainers = with maintainers; [ tesq0 ];
  };
}
```

lgpl2Only or lgpl2Plus?

File: pkgs/development/libraries/gtkextra/default.nix [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+  cairo,
+  atk,
+  pango,
+  libtiff,
+  libpng,
+  libjpeg
+}:
+
+let
+  minorVersion = "3.3";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gtkextra/default.nix:18]: Could we split this of the version? That would auto updates easier.

File: pkgs/applications/science/electronics/lepton-eda/default.nix (hunk 6 above)
Comment (tesq0) [reply pkgs/applications/science/electronics/lepton-eda/default.nix]: updated to gpl2Plus

File: pkgs/development/libraries/gtkextra/default.nix (hunk 8 above)
Comment (tesq0) [reply pkgs/development/libraries/gtkextra/default.nix:28]: yep it works

File: pkgs/development/libraries/gtkextra/default.nix [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+  version = "3.3.4";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/project/gtkextra/${lib.versions.majorMinor version}/${pname}-${version}.tar.gz";
+    sha256 = "1mpihbyzhv3ymfim93l9xnxmzhwyqdba5xb4rdn5vggdg25766v5";
+  };
+
+  preConfigure = ''
+    sed -i -e 's;/usr/bin/file;${file}/bin/file;g' ./configure
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gtkextra/default.nix:28]:
```suggestion
  postPatch = ''
    substituteInPlace configure \
      --replace "/usr/bin/file" "${file}/bin/file"
  '';
```

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise LGTM

File: pkgs/development/libraries/gtkextra/default.nix (hunk 11 above)
Comment (tesq0) [reply pkgs/development/libraries/gtkextra/default.nix]: Applied the change, thanks

**Pull request merged**